core/port.h

```diff
@@ -178,6 +179,8 @@ struct QueueStats {
   uint64_t packets;
   uint64_t dropped;  // Not all drivers support this for INC direction
   uint64_t bytes;    // It doesn't include Ethernet overhead
+  uint64_t drop_hist[bess::PacketBatch::kMaxBurst];
```

Comment:
(something like `rx_sent_hist` and `tx_drop_hist`, and `oneof` for the protobuf fields)

Reply:
Ah yes that's a good point. However, couldn't we simply have one histogram and let it be implicit whether or not the histogram is for recv or drops? It would be "stored" in whether the queue stats struct was in the PACKET_DIR_INC subarray or the inc field of the port stats struct right?

Reply:
That works too. I have another idea for direction-agnostic stats; for each direction we maintain 2 counters (something like `batch_size_hist.requested` and `batch_size_hist.actual`). This should give a better idea for diagnostics/monitoring. I am not particularly fond of the names or whether it should be an array of structs or a struct of arrays, so it's your call.

Reply:
Ah... it does not capture the histogram of drop counter. Hm....
Maybe we can add a third one, `batch_size_hist.failed`?

Reply:
(the client can calculate the drop_hist by subtracting actual from requested)